sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -772,9 +772,6 @@ object ColumnPruning extends Rule[LogicalPlan] {
       f.copy(child = prunedChild(child, f.references))
     case e @ Expand(_, _, child) if !child.outputSet.subsetOf(e.references) =>
       e.copy(child = prunedChild(child, e.references))
-    case s @ ScriptTransformation(_, _, _, child, _)
-        if !child.outputSet.subsetOf(s.references) =>
-      s.copy(child = prunedChild(child, s.references))
```

Comment:
Remove this since we don't need column pruning here

Reply:
Why don't we need the pruning?

Reply:
> Why don't we need the pruning?

it pruning input to child, child's output always transform's input, so the column pruning for script transform is not useful.

Reply:
Ah, okay. I remembered that `ScriptTransformation` always has a projected list of exprs in it's child now.